runtime/src/bank.rs

```diff
@@ -5646,6 +5617,38 @@ pub(crate) mod tests {
                     0xff, 0xff, 0xff, 0xff, 0xff, 0xff
                 ])
         );
+
+        fn check_no_remainder(partition_count: u64) -> bool {
```

Comment:
nits: hmm, how about renaming this to `should_cause_overflow`, the comment is great but a bit hard to grasp what this is doing at quick glance.